src/vaadin-details.ts

```diff
@@ -0,0 +1,128 @@
+import { html, css, customElement, property, PropertyValues } from 'lit-element';
+import { styleMap } from 'lit-html/directives/style-map.js';
+import { VaadinElement } from '@vaadin/element-base/vaadin-element.js';
+import { ControlStateMixin } from '@vaadin/control-state-mixin/control-state-mixin.js';
+
+/**
+ * `<vaadin-details>` is a Web Component creating an expandable panel
+ * similar to `<details>` HTML element.
+ *
+ * @element vaadin-details
+ *
+ * @slot - Slot fot the collapsible details content.
+ * @slot summary - Slot for the summary content.
+ *
+ * @attr {Boolean} focused - Set when the element is focused.
+ * @attr {Boolean} focus-ring - Set when the element is focused using the keyboard.
+ *
+ * @csspart toggle - The element used as indicator, can represent an icon.
+ * @csspart summary-content - The wrapper for the slotted summary content.
+ * @csspart summary - The element used to open and close collapsible content.
+ * @csspart content - The wrapper for the collapsible details content.
+ *
+ * @fires opened-changed - Fired when the `opened` property changes.
+ */
+@customElement('vaadin-details')
+export class VaadinDetails extends ControlStateMixin(VaadinElement) {
+  static is = 'vaadin-details';
+
+  static get version() {
+    return '1.0.1';
+  }
+
+  /**
+   * When true, the panel content is expanded and visible
+   */
+  @property({ type: Boolean, reflect: true }) opened = false;
+
+  static get styles() {
+    return css`
+      :host {
+        display: block;
+      }
+
+      :host([hidden]) {
+        display: none !important;
+      }
+
+      [part='content'] {
+        display: none;
+        overflow: hidden;
```

Comment:
Why is this `overflow: hidden` needed?

Reply:
Do you think `max-height` is enough? If so, then maybe I can remove this.

Reply:
But it’s `display: none;` and not rendered at all. Why is anything else needed?

Reply:
Removed. 

IIRC, the original code was most likely copied from [iron-collapse](https://github.com/PolymerElements/iron-collapse/blob/70e82a860beae7fdfd2b9e98d3612174b21a34f2/iron-collapse.js#L78).
We might change this later when we implement animated transitions.

Reply:
Since "block" is the default value for display here, the css could further be reduced to one rule:
```css
:host(:not([opened])) [part='content'] {
  display: none;
}
```